packages/plugin-page-view-tracking/src/page-view-tracking.ts

```diff
@@ -0,0 +1,75 @@
+import { IAmplitudeBrowser } from '@amplitude/analytics-browser';
+import {
+  BaseEvent,
+  BeforePlugin,
+  BrowserClient,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import { PluginCampaignTracker } from './plugin-campaign-tracker';
+import { PageTrackingBrowserOptions } from './typings/page-view-tracking';
+
+export class PageViewTrackingPlugin implements BeforePlugin {
+  name = 'page-view-tracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(
+    private readonly instance: BrowserClient | IAmplitudeBrowser,
+    private readonly options: PageTrackingBrowserOptions = {},
+  ) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
+    const attributionConfig = config.attribution;
+    const pageTrackingConfig = {
+      filter: attributionConfig?.trackPageViews ? 'onAttribution' : undefined,
```

Comment:
hmm i think we shouldnt use config.attribution to configure plugin. otherwise there are more combos. either purely core or purely plugin only

Reply:
Ideally each plugin should have its own config. This project is a bit different that this plugin depends on attribution configuration.

Reply:
but wouldn't the customer pass a separate config in the plugin anyway?